apps/src/applab/designElements/label.jsx

```diff
@@ -400,19 +422,41 @@ export default {
     element.style.height = size.height + 'px';
   },
 
+  _lastFitsExactly: {},
+
   /**
    * Returns whether this element perfectly fits its bounding size, if that is needed in onPropertyChange.
    */
-  beforePropertyChange: function(element, name) {
-    if (name !== 'text' && name !== 'fontSize') {
-      return null;
+  beforePropertyChange: function(element, name, batchChangeId) {
+    switch (name) {
+      case 'padding':
+      case 'text':
+      case 'fontFamily':
+      case 'fontSize': {
+        const {
+          batchId = -1,
+          previouslyFitExactly: batchPreviouslyFitExactly
+        } = this._lastFitsExactly;
```

Comment:
What does this do? I haven't seen this syntax before

Reply:
This is object destructuring from ES6. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring